src/NotificationServerFactory.php

```diff
@@ -149,6 +149,7 @@ private function validateRequest(RequestInterface $request)
             throw new NotificationValidationException('Header "SOAPAction" is missing from notification request.');
         }
         $body = $request->getBody()->getContents();
+        $request->getBody()->rewind();
```

Comment:
Had issue with that particular things in `ecphp/cas-lib`, since it's using streams. I hate those things.